### PR TITLE
Encoding tests cleanup

### PR DESCRIPTION
**Pull request by benaadams**
Follow up to #15514

### CODE REVIEW

File: src/System.Text.Encoding/tests/Encoder/EncoderConvert2.cs [hunk 1]
```diff
@@ -32,15 +32,34 @@ public override int GetBytes(char[] chars, int charIndex, int charCount, byte[]
...
         private const int c_SIZE_OF_ARRAY = 256;
         private readonly RandomDataGenerator _generator = new RandomDataGenerator();
-        #endregion
 
-        #region Positive Test Cases
-        // PosTest1: Call Convert to convert a arbitrary character array with UTF8 encoder
+        // Call Convert to convert a arbitrary character array with ASCII encoder
         [Fact]
-        public void PosTest1()
+        public void EncoderASCIIConvertRandomCharArray()
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/Encoder/EncoderConvert2.cs]: >     [](start = 0, length = 4)

looks we have many methods which has exactly same code and differ in just one thing. like this method and the next 2 methods are same and differ in the encoder. can we try to to make it one theory method and pass the encoder as data to it?

**Comment by tarekgh** [comment]
left one little comment, other than that LGTM. thank you!

**Comment by benaadams** [comment]
Converting some others in that class to Theory also

**Comment by benaadams** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

**Comment by benaadams** [comment]
@tarekgh PTAL

**Comment by tarekgh** [comment]
sorry, it looks I was not clear in my last comment. what I mean is the methods EncoderASCIIConvertRandomCharArray, EncoderUTF8ConvertRandomCharArray and EncoderUnicodeConvertRandomCharArray in EncoderConvert2.cs file has almost same code. so what I suggest is to have just one theory method and pass the needed data to it.

**Comment by benaadams** [comment]
@tarekgh made those changes also

**Comment by tarekgh** [comment]
@benaadams thanks again. looks good. I'll merge as soon we get the green checks. have a good weekend.

**Comment by benaadams** [comment]
`System.Net.Sockets.Tests.DualModeConnectAsync.ConnectAsyncV4IPEndPointToV6Host_Fails`

@dotnet-bot test Innerloop CentOS7.1 Release Build and Test

**Pull request merged**